File: FrontOne.Domain/Entities/VigenciaListaPrecioFruta.cs
namespace FrontOne.Domain.Entities;

// Proyección de sp_ListaPrecioFruta_ObtenerFechas: una combinación única de fecha +
// productor-o-general, navegable desde el buscador. No es una fila de la tabla en sí.
public class VigenciaListaPrecioFruta
{
    public DateTime FechaInicio { get; set; }
    public int? ProductorId { get; set; }
    public string? ProductorNombre { get; set; }
}

File: FrontOne.Android/domain/src/main/kotlin/com/frontone/android/domain/model/VigenciaListaPrecioFruta.kt
package com.frontone.android.domain.model

import java.time.LocalDate

/**
 * Una fecha (+ productor opcional, `null` = lista general) con precios ya guardados en
 * `Acopio.ListaPrecioFruta` — universo de vigencias que el selector "Cargar Precios" de
 * la Calculadora ofrece para elegir. Ver `Acopio.sp_ListaPrecioFruta_ObtenerFechas`.
 */
data class VigenciaListaPrecioFruta(
    val fecha: LocalDate,
    val productorId: Int?,
    val productorNombre: String?
)
